js/pro/poloniex.js

```diff
@@ -0,0 +1,886 @@
+                    const price = this.safeNumber (ask, 0);
+                    const amount = this.safeNumber (ask, 1);
+                    orderbook['asks'].store (price, amount);
+                }
+            }
+            orderbook['timestamp'] = timestamp;
+            orderbook['datetime'] = this.iso8601 (timestamp);
+            client.resolve (orderbook, messageHash);
+        }
+    }
+
+    handleBalance (client, message) {
+        //
+        //    {
+        //       "channel": "balances",
+        //       "data": [
+        //            {
+        //                "changeTime": 1657312008411,
+        //                "accountId": "1234",
+        //                "accountType": "SPOT",
+        //                "eventType": "place_order",
+        //                "available": "9999999983.668",
+        //                "currency": "BTC",
+        //                "id": 60018450912695040,
+        //                "userId": 12345,
+        //                "hold": "16.332",
+        //                "ts": 1657312008443
+        //            }
+        //        ]
+        //    }
+        //
+        const data = this.safeValue (message, 'data', []);
+        const messageHash = 'balances';
+        this.balance = this.parseWsBalance (data);
+        client.resolve (this.balance, messageHash);
+    }
+
+    parseWsBalance (response) {
+        //
+        //    [
+        //        {
+        //            "changeTime": 1657312008411,
+        //            "accountId": "1234",
+        //            "accountType": "SPOT",
+        //            "eventType": "place_order",
+        //            "available": "9999999983.668",
+        //            "currency": "BTC",
+        //            "id": 60018450912695040,
+        //            "userId": 12345,
+        //            "hold": "16.332",
+        //            "ts": 1657312008443
+        //        }
+        //    ]
+        //
+        const firstBalance = this.safeValue (response, 0, {});
+        const timestamp = this.safeInteger (firstBalance, 'ts');
+        const result = {
+            'info': response,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+        };
+        for (let i = 0; i < response.length; i++) {
+            const balance = this.safeValue (response, i);
+            const currencyId = this.safeString (balance, 'currency');
+            const code = this.safeCurrencyCode (currencyId);
+            const newAccount = this.account ();
+            newAccount['free'] = this.safeString (balance, 'available');
+            newAccount['used'] = this.safeString (balance, 'hold');
+            result[code] = newAccount;
+        }
+        return this.safeBalance (result);
+    }
+
+    handleMessage (client, message) {
+        const type = this.safeString (message, 'channel');
+        const methods = {
+            'candles_minute_1': this.handleOHLCV,
```

Comment:
I think it would be easier to do something like:

```Javascript
if (type.indexOf('candles_') {
 ....
}
```
Instead of manually listing every possible timeframe

Reply:
There's not a lot of benefit to this, you end up needing to check if type is defined, and it's less lines of code but it's more complicated